File: src/vb/Samples/ApiSamples/Samples/Application/AddObjectsToSelectSet.vb
Imports SolidEdgeFramework.Extensions 'SolidEdge.Community.dll
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text

Namespace ApiSamples.Application
	''' <summary>
	''' Adds objects from the active document to the document select set.
	''' </summary>
	Friend Class AddObjectsToSelectSet
		Private Shared Sub RunSample(ByVal breakOnStart As Boolean)
			If breakOnStart Then
				System.Diagnostics.Debugger.Break()
			End If

			Dim application As SolidEdgeFramework.Application = Nothing
			Dim document As SolidEdgeFramework.SolidEdgeDocument = Nothing
			Dim selectSet As SolidEdgeFramework.SelectSet = Nothing
			Dim assemblyDocument As SolidEdgeAssembly.AssemblyDocument = Nothing
			Dim asmRefPlanes As SolidEdgeAssembly.AsmRefPlanes = Nothing
			Dim draftDocument As SolidEdgeDraft.DraftDocument = Nothing
			Dim sheet As SolidEdgeDraft.Sheet = Nothing
			Dim drawingViews As SolidEdgeDraft.DrawingViews = Nothing
			Dim partDocument As SolidEdgePart.PartDocument = Nothing
			Dim sheetMetalDocument As SolidEdgePart.SheetMetalDocument = Nothing
			Dim edgeBarFeatures As SolidEdgePart.EdgebarFeatures = Nothing

			Try
				' Register with OLE to handle concurrency issues on the current thread.
				SolidEdgeCommunity.OleMessageFilter.Register()

				' Connect to or start Solid Edge.
				application = SolidEdgeCommunity.SolidEdgeInstall.Start()

				' Get a reference to the active document.
				document = application.GetActiveDocument(Of SolidEdgeFramework.SolidEdgeDocument)(False)

				If document IsNot Nothing Then
					' Determine document type.
					Select Case document.Type
						Case SolidEdgeFramework.DocumentTypeConstants.igAssemblyDocument
							assemblyDocument = DirectCast(document, SolidEdgeAssembly.AssemblyDocument)
							asmRefPlanes = assemblyDocument.AsmRefPlanes
							selectSet = assemblyDocument.SelectSet

							For i As Integer = 1 To asmRefPlanes.Count
								selectSet.Add(asmRefPlanes.Item(i))
							Next i

						Case SolidEdgeFramework.DocumentTypeConstants.igDraftDocument
							draftDocument = DirectCast(document, SolidEdgeDraft.DraftDocument)
							sheet = draftDocument.ActiveSheet
							drawingViews = sheet.DrawingViews
							selectSet = draftDocument.SelectSet

							For i As Integer = 1 To drawingViews.Count
								draftDocument.SelectSet.Add(drawingViews.Item(i))
							Next i

						Case SolidEdgeFramework.DocumentTypeConstants.igPartDocument
							partDocument = DirectCast(document, SolidEdgePart.PartDocument)
							edgeBarFeatures = partDocument.DesignEdgebarFeatures
							selectSet = partDocument.SelectSet

							For i As Integer = 1 To edgeBarFeatures.Count
								partDocument.SelectSet.Add(edgeBarFeatures.Item(i))
							Next i

						Case SolidEdgeFramework.DocumentTypeConstants.igSheetMetalDocument
							sheetMetalDocument = DirectCast(document, SolidEdgePart.SheetMetalDocument)
							edgeBarFeatures = sheetMetalDocument.DesignEdgebarFeatures
							selectSet = sheetMetalDocument.SelectSet

							For i As Integer = 1 To edgeBarFeatures.Count
								partDocument.SelectSet.Add(edgeBarFeatures.Item(i))
							Next i
					End Select
				Else
					Throw New System.Exception(Resources.NoActiveDocument)
				End If
			Catch ex As System.Exception
				Console.WriteLine(ex.Message)
			Finally
				SolidEdgeCommunity.OleMessageFilter.Unregister()
			End Try
		End Sub
	End Class
End Namespace

File: cs/SolidEdge/ApiDemos/ApiDemos/Samples/Application/AddObjectsToSelectSet.cs
using SolidEdgeCommunity.Extensions; // Enabled extension methods from SolidEdge.Community.dll
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace ApiDemos.Application
{
    /// <summary>
    /// Adds objects from the active document to the document select set.
    /// </summary>
    class AddObjectsToSelectSet
    {
        static void RunSample(bool breakOnStart)
        {
            if (breakOnStart) System.Diagnostics.Debugger.Break();

            SolidEdgeFramework.Application application = null;
            SolidEdgeFramework.SolidEdgeDocument document = null;
            SolidEdgeFramework.SelectSet selectSet = null;
            SolidEdgeAssembly.AssemblyDocument assemblyDocument = null;
            SolidEdgeAssembly.AsmRefPlanes asmRefPlanes = null;
            SolidEdgeDraft.DraftDocument draftDocument = null;
            SolidEdgeDraft.Sheet sheet = null;
            SolidEdgeDraft.DrawingViews drawingViews = null;
            SolidEdgePart.PartDocument partDocument = null;
            SolidEdgePart.SheetMetalDocument sheetMetalDocument = null;
            SolidEdgePart.EdgebarFeatures edgeBarFeatures = null;

            try
            {
                // Register with OLE to handle concurrency issues on the current thread.
                SolidEdgeCommunity.OleMessageFilter.Register();

                // Connect to or start Solid Edge.
                application = SolidEdgeCommunity.SolidEdgeUtils.Start();

                // Get a reference to the active document.
                document = application.GetActiveDocument<SolidEdgeFramework.SolidEdgeDocument>(false);

                if (document != null)
                {
                    // Determine document type.
                    switch (document.Type)
                    {
                        case SolidEdgeFramework.DocumentTypeConstants.igAssemblyDocument:
                            assemblyDocument = (SolidEdgeAssembly.AssemblyDocument)document;
                            asmRefPlanes = assemblyDocument.AsmRefPlanes;
                            selectSet = assemblyDocument.SelectSet;

                            for (int i = 1; i <= asmRefPlanes.Count; i++)
                            {
                                selectSet.Add(asmRefPlanes.Item(i));
                            }

                            break;
                        case SolidEdgeFramework.DocumentTypeConstants.igDraftDocument:
                            draftDocument = (SolidEdgeDraft.DraftDocument)document;
                            sheet = draftDocument.ActiveSheet;
                            drawingViews = sheet.DrawingViews;
                            selectSet = draftDocument.SelectSet;

                            for (int i = 1; i <= drawingViews.Count; i++)
                            {
                                draftDocument.SelectSet.Add(drawingViews.Item(i));
                            }

                            break;
                        case SolidEdgeFramework.DocumentTypeConstants.igPartDocument:
                            partDocument = (SolidEdgePart.PartDocument)document;
                            edgeBarFeatures = partDocument.DesignEdgebarFeatures;
                            selectSet = partDocument.SelectSet;

                            for (int i = 1; i <= edgeBarFeatures.Count; i++)
                            {
                                partDocument.SelectSet.Add(edgeBarFeatures.Item(i));
                            }

                            break;
                        case SolidEdgeFramework.DocumentTypeConstants.igSheetMetalDocument:
                            sheetMetalDocument = (SolidEdgePart.SheetMetalDocument)document;
                            edgeBarFeatures = sheetMetalDocument.DesignEdgebarFeatures;
                            selectSet = sheetMetalDocument.SelectSet;

                            for (int i = 1; i <= edgeBarFeatures.Count; i++)
                            {
                                partDocument.SelectSet.Add(edgeBarFeatures.Item(i));
                            }
                            break;
                    }
                }
                else
                {
                    throw new System.Exception(Resources.NoActiveDocument);
                }
            }
            catch (System.Exception ex)
            {
                Console.WriteLine(ex.Message);
            }
            finally
            {
                SolidEdgeCommunity.OleMessageFilter.Unregister();
            }
        }
    }
}
